package/gluon-neighbour-info/src/gluon-neighbour-info.c

```diff
@@ -206,10 +208,20 @@ int main(int argc, char **argv) {
     }
 
   if (request_string == NULL) {
-    fprintf(stderr, "No request string supplied");
+    fprintf(stderr, "No request string supplied\n");
     exit(EXIT_FAILURE);
   }
 
+  if (client_addr.sin6_port == htons(0)) {
+	fprintf(stderr, "No port supplied\n");
+	exit(EXIT_FAILURE);
+  }
+
+  if (!destination_set) {
```

Comment:
You don't need to introduce a new variable, you can just keep the ip6addr_any default value and check with IN6_IS_ADDR_UNSPECIFIED().